### DISCUSSION TITLE
Tests failed to load after upgrading to 5.1.1

### DISCUSSION THREAD

**Discussion by mike-4040** [chaijs/chai#1627] [Support]
## Environment
1. Node 20.11.1

2. Dependencies
```
    "chai": "^5.1.1",
    "mocha": "^10.4.0",
    "ts-node": "^10.9.2",
    "typescript": "^5.4.5",
```

3. Tests configuration
 - .mocharc file
 ```
# .mocharc
{
  "$schema": "https://json.schemastore.org/mocharc",
  "reporter": "spec",
  "require": ["ts-node/register", "src/testUtils/global-mocha-fixture.ts"],
}
```
- test file
```
# ./src/sample.test.ts
import { expect } from 'chai';

describe('Simple Test', () => {
  it('should return true', () => {
    expect(true).to.be.true;
  });
});
```

- running test `npx ./node_modules/mocha/bin/mocha.js ./src/sample.test.ts`

##  Issue
After upgrading chai 4.4.1 => 5.1.1 consistently getting Error:

```
 Exception during run: TypeError: Unknown file extension ".ts" for ./src/sample.test.ts
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at async ModuleLoader.load (node:internal/modules/esm/loader:409:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:45)
    at async link (node:internal/modules/esm/module_job:76:21) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}
```

**Comment by 43081j** [comment]
you need to set `loader` for ESM in mocha:

https://gist.github.com/43081j/78ce1392abb5043b02a29355006880a5#file-mocharc-json

afaik, you can also have both `loader` and `require` set (in case somehow you have a mixture of cjs/esm)

**Comment by sgpinkus** [comment]
I have the same issue after upgrade from v4 to v5.

> you need to set loader for ESM in mocha ..

If that's the case shouldn't this issue also crop up in chai v4? Besides, we're already using the typescript loader.

**Reply by 43081j** [reply]
The typescript loader supports commonjs and es modules

It's up to you to tell node which you want. If you use `ts-node/register`, that will add typescript support to commonjs modules. You must use `ts-node/esm` for esm support (and `loader` instead of `require `. Basically the previously explained example)

**Reply by sgpinkus** [reply]
Thanks for the reply. Tried using the config from the example:

```

{
  "loader": "ts-node/esm",
  "extensions": ["ts"],
  "spec": [
    "tests/*.ts"
  ]
}
```

Didn't work, with or with `type: module` in package.json.

What changed between v4 and v5?

**Reply by 43081j** [reply]
The example certainly works. Can you provide a reproduction of where you're seeing it fail? 

chai is now esm only. Any problems loading it are very unlikely to be chai's fault. So an example will help us a lot to debug your problem
